R/permissions.R

```diff
@@ -0,0 +1,216 @@
+                                    data)
+  }
+  
+  return(permissions %>% 
+           dplyr::arrange(.data$id,
+                          .data$permission))
+}
+
+#' Is WebAPI using authentication/authorization
+#'
+#' @template baseUrl 
+#'
+#' @return TRUE or FALSE
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' securityEnabled(baseUrl = "http://server.org:80/WebAPI")
+#' }
+securityEnabled <- function(baseUrl) {
+  .checkBaseUrl(baseUrl)
+  url <- paste0(baseUrl, "/info")
+  response <- httr::GET(url)
+  httr::content(response)$configuration$security$enabled
+}
+
+
+#' Obtains users for a specific WebAPI user role. Requires security enabled WebAPI
+#' \lifecycle{experimental}
+#' @details
+#' For a security enabled WebAPI, obtains the user names for a specified Atlas user role. Role Id is
+#' set in WebAPI sec_role table
+#'
+#' @template BaseUrl
+#' @param roleId   The role id as defined in WebAPI sec_role table
+#' @return
+#' A data frame of user information for the WebAPI role.
+#'
+#' @examples
+#' \dontrun{
+#' getUsersFromRole(baseUrl = "http://server.org:80/WebAPI", roleId = 10)
+#' }
+#' @export
+getUsersFromRole <- function(baseUrl, roleId) {
+  
+  .checkBaseUrl(baseUrl)
+  url <- sprintf("%s/role/%s/users", baseUrl, roleId)
+  request <- .GET(url)
+  httr::stop_for_status(request)
+  
+  json <- .GET(url)
+  users <- httr::content(json)
+  
+  result <- lapply(users, function(u) {
+    data.frame(id = u$id, login = u$login, name = u$name)
+  })
+  return(do.call(rbind, result) %>% 
+           dplyr::tibble())
+}
+
+
+#' Assign cohort definition permissions to user. \lifecycle{experimental}
+#'
+#' @details
+#' The function will call the webservice to assign permissions to a cohort definition for the given userId.
+#'
+#' @template baseUrl
+#' @param cohortId    The ID of the cohort definition
+#' @param userId      The id to grant permission. You can find out the id from userName 
+#'                    as getUserIdFromUserName(baseUrl = baseUrl, userName = "myId")
+#' @param permission  The permission type to assign.  Can be "READ" or "WRITE". Defaults to "WRITE".
+#'
+#' @examples
+#' \dontrun{
+#' setCohortPermission(baseUrl, cohortId, userId = getUserIdFromUserName(baseUrl = baseUrl, userName = "myId"), permission = "READ")
+#' }
+#' @export
+setCohortPermission <-
+  function(baseUrl, cohortId, userId, permission = "READ") {
+    .checkBaseUrl(baseUrl)
+    if (!securityEnabled(baseUrl)) {
+      message("Security is not enabled so permissions cannot be set.")
+      return(invisible(NULL))
+    }
+    
+    if (getWebApiVersion(baseUrl) < "2.10") {
+      message("setCohortPermission can only be used with WebAPI version 2.10 and later")
+      return(invisible(NULL))
+    }
+    
+    if (!permission %in% c("READ", "WRITE")) {
+      stop(paste0(
+        "Invalid Permission Type:",
+        permission,
+        ".  Valid types are READ or WRITE."
+      ))
+    }
+    
+    url <-
```

Comment:
http://webapidoc.ohdsi.org/index.html#417537417

POST http://www.example.com/permission/access/entityType/entityId/role/roleId/

is userId the same as roleId -- i dont think so.

e.g. roleId =1 is public. I can assign WRITE permissions to roleId 1!

Reply:
This morning i was able to assign WRITE permission (cant do READ yet) to another user using this funcitn